src/client/Microsoft.Identity.Client/Instance/Discovery/NetworkMetadataProvider.cs

```diff
@@ -72,7 +72,7 @@ private void CacheInstanceDiscoveryMetadata(InstanceDiscoveryResponse instanceDi
             return discoveryResponse;
         }
 
-        private async Task<InstanceDiscoveryResponse> SendInstanceDiscoveryRequestAsync(
+        private Task<InstanceDiscoveryResponse> SendInstanceDiscoveryRequestAsync(
```

Comment:
Not sure about this one. Lines 79 - 84 could throw exception and the ex behavior is subtly different. See https://blog.stephencleary.com/2016/12/eliding-async-await.html

Reply:
yeah i am aware of the advice. and after years of eliding all my async on all my projects (https://www.nuget.org/profiles/simoncropp), i am yet to be negatively impact by the diff exception handling

also note that the majority of MS codebases also elide all their asyncs